threatgrid/ctia
# Filter incidents by tactics

**Epic** https://github.com/advthreat/iroh/issues/6969
Related https://github.com/advthreat/iroh/issues/7206

Incidents are filterable by the new `tactics` field. However, we need to add tests so QA can verify the mappings were actually refreshed since `tactics` was added.



<a name="qa">[§](#qa)</a> QA
============================


Assumption: all `TA####` ids below have never been assigned to other incidents in TEST.

Note: The "search route" below is: `GET https://private.intel.test.iroh.site/ctia/incident/search` and the example query params are not yet url-encoded.

1. Create incident1 with field `tactics: ["TA1234", "TA4321"]` in [private intel](https://private.intel.test.iroh.site/index.html#/Incident/post_ctia_incident)
2. Create incident2 with field `tactics: ["TA1234", "TA8888"]` in [private intel](https://private.intel.test.iroh.site/index.html#/Incident/post_ctia_incident)
3. Verify the search route returns just incident1 and incident2 with query-params: `query=tactics:("TA1234")`
4. Verify the search route returns just incident1 with query-params: `query=tactics:("TA4321")`
5. Verify the search route returns just incident2 with query-params: `query=tactics:("TA8888")`
6. Verify the search route returns just incident1 and incident2 with query-params: `query=tactics:("TA4321"||"TA8888")`






<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================



```
intern: Filter incidents by tactics
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================
